### PR TITLE
feat: prepare v1

### PR DESCRIPTION
**Pull request by pixelass**
- add base file logic
- save prompt config
- error logs
- rename goal to prompt and neg to negativePrompt
- move fileWriter to base
- adjust webpack
- introduce seed (pseudo)
- better JS validation
- add several optional dependencies

closes #2

### CODE REVIEW

File: base.js [hunk 1]
```diff
@@ -7,32 +7,35 @@ import { Configuration, OpenAIApi } from "openai";
...
   -t, --temperature    Set the temperature for the generated code. Default is 0.2.
-  -c, --clean          Set to true if you want to remove any previously generated code.
+  -c, --clean          Set to true if you want to remove any previously generated code.  Default is false.
   -m, --model          Set the model to use for generating the code. Default is "gpt-3.5-turbo".
+  -n, --negativePrompt Set the negative prompt. Default is "".
+  -s, --seed           Set the seed. Default is -1.
 
 Examples
-  $ node generation-000.js -G "console based mandelbrot set ascii" -g 3 -p "Node.js developer, creative" -c
+  $ node generation-000.js -p "mandelbrot set" -g 3 -P "Node.js developer, creative" -c
```
Comment (TimPietrusky) [inline base.js:29]: Get rid of the persona, as we already have those as the default

```suggestion
  $ node generation-000.js -p "mandelbrot set" -g 3 -c
```

File: base.js [hunk 2]
```diff
@@ -7,32 +7,35 @@ import { Configuration, OpenAIApi } from "openai";
...
+  $ node generation-000.js -p "matrix code" -g 10 -c -s 123456789
```
Comment (TimPietrusky) [inline base.js:30]: Maybe use a negative prompt here as the example to have some more variation or use a total different persona

**Review by TimPietrusky — CHANGES_REQUESTED** [review]
LGTM, I only have two have additions. 

But will test the code now also.

File: base.js (hunk 1 above)
Comment (pixelass) [reply base.js]: this is valid since it showcases a persona that is different from the default

File: base.js (hunk 2 above)
Comment (pixelass) [reply base.js]: adjusted. I added another example

**Review by TimPietrusky — APPROVED** [review]
LGTM, thanks!

**Pull request merged**